backend/lib/endpoints/feedback.js

```diff
@@ -0,0 +1,36 @@
+const { createFeedbackSchema } = require("./schema/feedback");
+
+/*
+ * /api/feedback
+ */
+async function routes(app) {
+  const Feedback = app.mongo.model("Feedback");
+
+  app.post("/", { schema: createFeedbackSchema }, async (req, reply) => {
+    const { userId } = req.body;
+
+    if (userId) {
+      const [err, userFeedback] = await app.to(Feedback.find({ userId }));
+      if (userFeedback[0] || err) {
+        throw app.httpErrors.internalServerError();
+      }
+    }
+
+    const [err] = await app.to(
+      new Feedback({
+        ...req.body,
+        ipAddress: req.ip,
+      }).save(),
+    );
+
+    if (err) throw app.httpErrors.internalServerError();
```

Comment:
Yeah I think this is much better, you could even `throw error` with that generic catch handler it would just return a 500 with generic message too, but maybe better to log it. And if we want we could still handle special cases here differently
so I think something like this would give us useful error info
```
if (err) {
  req.log.error("Failed submitting feedback", { err }); 
  throw app.httpErrors.internalServerError();
}
``